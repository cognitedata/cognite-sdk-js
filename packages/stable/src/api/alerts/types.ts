// Copyright 2020 Cognite AS

import {
  Metadata,
  MetadataPatch,
  SinglePatchRequired,
  SinglePatchRequiredString,
  SinglePatchString,
  Timestamp,
  DateRange,
} from '@cognite/sdk';
import {
  CogniteExternalId,
  CogniteInternalId,
  ExternalId,
  FilterQuery,
  InternalId,
} from '@cognite/sdk-core';

export * from '@cognite/sdk';

// This file is here mostly to allow apis to import { ... } from '../../types';
// Overriding types should probably be done in their respective API endpoint files, where possible

export type AlertStatus = 'FIRING' | 'PENDING';

export const AlertStatus = {
  FIRING: 'FIRING' as AlertStatus,
  PENDING: 'PENDING' as AlertStatus,
};

export interface AlertFilter {
  channelIds?: CogniteInternalId[];
  channelExternalIds?: CogniteExternalId[];
  closed?: boolean;
  startTime?: DateRange;
  status?: AlertStatus[];
}

export interface AlertFilterQuery extends FilterQuery {
  filter?: AlertFilter;
}

export interface AlertDeduplicationRuleCreate {
  mergeInterval?: string;
  activationInterval?: string;
}

export interface AlertRulesCreate {
  deduplication: AlertDeduplicationRuleCreate;
}

export interface AlertDeduplicationRule {
  mergeInterval?: string;
  activationInterval?: string;
}

export interface AlertRules {
  deduplication: AlertDeduplicationRule;
}

export interface AlertCreate {
  externalId?: CogniteExternalId;
  timestamp?: Timestamp;
  channelId?: CogniteInternalId;
  channelExternalId?: CogniteExternalId;
  source: string;
  value?: string;
  level?: 'DEBUG' | 'INFO' | 'WARNING' | 'ERROR' | 'CRITICAL';
  metadata?: Metadata;
}

export interface Alert {
  id: CogniteInternalId;
  externalId?: CogniteExternalId;
  timestamp: Timestamp;
  channelId: CogniteInternalId;
  channelExternalId?: CogniteExternalId;
  source: string;
  value?: string;
  level?: 'DEBUG' | 'INFO' | 'WARNING' | 'ERROR' | 'CRITICAL';
  metadata?: Metadata;
  acknowledged: boolean;
  closed: boolean;
  status: AlertStatus;
}

export interface ChannelCreate {
  externalId?: CogniteExternalId;
  parentExternalId?: CogniteExternalId;
  parentId?: CogniteInternalId;
  name: string;
  description?: string;
  metadata?: Metadata;
  alertRules?: AlertRulesCreate;
}

export interface Channel {
  id: CogniteInternalId;
  externalId?: CogniteExternalId;
  parentExternalId?: CogniteExternalId;
  parentId?: CogniteInternalId;
  name: string;
  description?: string;
  metadata?: Metadata;
  alertRules?: AlertRules;
}

export interface ChannelFilter {
  ids?: CogniteInternalId[];
  externalIds?: CogniteExternalId[];
  parentIds?: CogniteInternalId[];
  metadata?: Metadata;
}

export interface ChannelFilterQuery extends FilterQuery {
  filter?: ChannelFilter;
}

export interface ChannelPatch {
  update: {
    externalId?: SinglePatchString;
    name?: SinglePatchRequiredString;
    description?: SinglePatchString;
    parentId?: SinglePatchRequired<number>;
    metadata?: MetadataPatch;
  };
}

export interface ChannelChangeById extends ChannelPatch, InternalId {}

export interface ChannelChangeByExternalId extends ChannelPatch, ExternalId {}

export type ChannelChange = ChannelChangeById | ChannelChangeByExternalId;

export interface SubscriberCreate {
  externalId?: CogniteExternalId;
  metadata?: Metadata;
  email: string;
}

export interface SubscriberFilter {
  externalIds?: CogniteExternalId[];
  ids?: CogniteInternalId[];
  metadata?: Metadata;
  email?: string;
}

export interface SubscriberFilterQuery extends FilterQuery {
  filter?: SubscriberFilter;
}

export interface Subscriber {
  id: CogniteInternalId;
  externalId?: CogniteExternalId;
  metadata?: Metadata;
  email: string;
}

export interface SubscriberCreate {
  externalId?: CogniteExternalId;
  metadata?: Metadata;
  email: string;
}

export interface SubscriptionCreate {
  externalId?: CogniteExternalId;
  channelId?: CogniteInternalId;
  channelExternalId?: CogniteExternalId;
  subscriberId?: CogniteInternalId;
  subscriberExternalId?: CogniteExternalId;
  metadata?: Metadata;
}

export interface SubscriptionFilter {
  externalIds?: CogniteExternalId[];
  channelIds?: CogniteInternalId[];
  channelExternalIds?: CogniteExternalId[];
  subscriberIds?: CogniteInternalId[];
  subscriberExternalIds?: CogniteExternalId[];
  metadata?: Metadata;
}

export interface SubscriptionFilterQuery extends FilterQuery {
  filter?: SubscriptionFilter;
}

export interface SubscriptionDelete {
  externalId?: CogniteExternalId;
  channelExternalId?: CogniteExternalId;
  channelId?: CogniteInternalId;
  subscriberId?: CogniteInternalId;
  subscriberExternalId?: CogniteExternalId;
}

export interface Subscription {
  externalId?: CogniteExternalId;
  channelId: CogniteInternalId;
  channelExternalId?: CogniteExternalId;
  subscriberId: CogniteInternalId;
  subscriberExternalId?: CogniteExternalId;
  metadata?: Metadata;
}
