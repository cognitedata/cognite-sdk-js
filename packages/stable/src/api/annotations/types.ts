// Copyright 2022 Cognite AS

import {
  CogniteInternalId,
  FilterQuery,
  IdEither,
  InternalId,
  SetField,
} from '@cognite/sdk';

export type AnnotatedResourceType = 'file';
export type AnnotationStatus = 'suggested' | 'approved' | 'rejected';

// TODO [CXT-463] Use annotation-types package definitions
export type AnnotationType = string;
export type AnnotationPayload = object;

export interface AnnotationModel extends AnnotationCreate {
  id: CogniteInternalId;
  createdTime: Date;
  lastUpdatedTime: Date;
}

export interface AnnotationCreate extends AnnotationSuggest {
  status: AnnotationStatus;
}

export interface AnnotationSuggest {
  annotatedResourceType: AnnotatedResourceType;
  annotatedResourceId: CogniteInternalId;
  annotationType: AnnotationType;
  creatingApp: string;
  creatingAppVersion: string;
  creatingUser: string | null;
  data: AnnotationPayload;
}

export interface AnnotationChangeById extends InternalId, AnnotationUpdate {}

export interface AnnotationUpdate {
  update: {
    annotationType?: SetField<AnnotationType>;
    data?: SetField<AnnotationPayload>;
    status?: SetField<AnnotationStatus>;
  };
}

export interface AnnotationFilterRequest
  extends AnnotationFilter,
    FilterQuery {}

export interface AnnotationFilter {
  filter: AnnotationFilterProps;
}

export interface AnnotationFilterProps {
  annotatedResourceType: AnnotatedResourceType;
  annotatedResourceIds: IdEither[];
  annotationType?: AnnotationType;
  creatingApp?: string;
  creatingAppVersion?: string;
  creatingUser?: string | null;
  status?: AnnotationStatus;
  data?: AnnotationPayload;
}
