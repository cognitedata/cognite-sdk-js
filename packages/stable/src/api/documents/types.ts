import { CogniteExternalId, CogniteInternalId } from '@cognite/sdk-core';

export type DocumentSearchRequest = DocumentFilterOption &
  DocumentSearch &
  DocumentAggregates &
  DocumentSort &
  DocumentSearchLimit;

export interface DocumentFilterOption {
  filter?: DocumentFilter;
}

export interface DocumentSearch {
  search?: { query?: string; highlight?: boolean };
}

export interface DocumentAggregates {
  aggregates?: DocumentCountAggregate[];
}

export interface DocumentSort {
  sort?: DocumentSortItem[];
}

export type DocumentSortOrder = 'asc' | 'desc';

export type DocumentSortItem = DocumentFilterProperty & {
  order: DocumentSortOrder;
};

export interface DocumentSearchLimit {
  limit?: number;
}

export interface Highlight {
  name?: string[];
  content?: string[];
}

export interface Document {
  id: CogniteInternalId;
  externalId?: CogniteExternalId;
  title?: string;
  author?: string;
  createdTime?: DocumentTimestamp;
  modifiedTime?: DocumentTimestamp;
  lastIndexedTime?: DocumentTimestamp;
  mimeType?: string;
  extension?: string;
  pageCount?: number;
  type?: string;
  language?: string;
  truncatedContent?: string;
  assetIds?: CogniteInternalId[];
  labels?: DocumentLabel[];
  sourceFile: DocumentSourceFile;
  geoLocation?: DocumentGeoJsonGeometry;
}

export interface DocumentLimit {
  limit?: number;
}

export type DocumentFilter =
  | {
      or: DocumentFilter[];
    }
  | {
      and: DocumentFilter[];
    }
  | {
      not: DocumentFilter;
    }
  | {
      equals: DocumentFilterPropertyAndValue;
    }
  | {
      prefix: DocumentFilterPropertyAndValue;
    }
  | {
      in: DocumentFilterPropertyAndValues;
    }
  | {
      containsAny: DocumentFilterPropertyAndValues;
    }
  | {
      containsAll: DocumentFilterPropertyAndValues;
    }
  | {
      range: DocumentFilterPropertyAndRangeValues;
    }
  | {
      exists: DocumentFilterProperty;
    }
  | {
      geojsonIntersects: DocumentFilterPropertyAndGeometry;
    }
  | {
      geojsonDisjoint: DocumentFilterPropertyAndGeometry;
    }
  | {
      geojsonWithin: DocumentFilterPropertyAndGeometry;
    };

export type DocumentFilterProperty = { property: string[] };

export type DocumentFilterValue = {
  value: string | number | boolean | DocumentLabel;
};

export type DocumentFilterValues = {
  values: string[] | number[] | boolean[] | DocumentLabel[];
};

export type DocumentFilterRangeValues = {
  gt?: number;
  lt?: number;
  gte?: number;
  lte?: number;
};

export type DocumentFilterGeometry = { geometry: DocumentGeoJsonGeometry };

export type DocumentFilterPropertyAndValue = DocumentFilterProperty &
  DocumentFilterValue;

export type DocumentFilterPropertyAndValues = DocumentFilterProperty &
  DocumentFilterValues;

export type DocumentFilterPropertyAndRangeValues = DocumentFilterProperty &
  DocumentFilterRangeValues;

export type DocumentFilterPropertyAndGeometry = DocumentFilterProperty &
  DocumentFilterGeometry;

export type DocumentAggregateType = 'count';

export interface DocumentCountAggregate {
  name: string;
  aggregate: DocumentAggregateType;
  groupBy?: DocumentFilterProperty[];
}

export type DocumentTimestamp = number;

export interface DocumentSourceFile {
  name: string;
  directory?: string;
  source?: string;
  mimeType?: string;
  size?: number;
  assetIds?: CogniteInternalId[];
  labels?: DocumentLabel[];
  geoLocation?: DocumentGeoJsonGeometry;
  datasetId?: CogniteInternalId;
  securityCategories?: number[];
  metadata?: Record<string, string>;
}

export interface DocumentGeoJsonGeometry {
  type?: DocumentShapeType;
  coordinates?: DocumentShapeCoordinates;
  geometries?: GeometryCollection[];
}

export interface DocumentLabel {
  externalId: CogniteExternalId;
}

export type DocumentShapeType = string;

export type DocumentShapeCoordinates = any[];

export interface GeometryCollection {
  type?: DocumentShapeType;
  coordinates?: DocumentShapeCoordinates;
}

export interface DocumentSearchResponse {
  items: { highlight?: Highlight; item: Document }[];
  aggregates?: {
    name: string;
    groups: {
      group: DocumentFilterPropertyAndValue[];
      count: number;
    }[];
    total: number;
  }[];
}
