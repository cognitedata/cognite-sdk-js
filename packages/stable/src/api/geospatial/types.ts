import { InternalId, Limit, ExternalId } from '@cognite/sdk-core';

import { Geometry, GeoJSON } from 'geojson';

export { GeoJSON };

export type GeometryType = 'WKT' | 'GEOJSON';

type GeometryPropertyType =
  | 'GEOMETRY'
  | 'POINT'
  | 'LINESTRING'
  | 'POLYGON'
  | 'MULTIPOINT'
  | 'MULTILINESTRING'
  | 'MULTIPOLYGON'
  | 'GEOMETRYCOLLECTION'
  | 'GEOMETRYZ'
  | 'POINTZ'
  | 'LINESTRINGZ'
  | 'POLYGONZ'
  | 'MULTIPOINTZ'
  | 'MULTILINESTRINGZ'
  | 'MULTIPOLYGONZ'
  | 'GEOMETRYM'
  | 'POINTM'
  | 'LINESTRINGM'
  | 'POLYGONM'
  | 'MULTIPOINTM'
  | 'MULTILINESTRINGM'
  | 'MULTIPOLYGONM'
  | 'GEOMETRYCOLLECTIONM'
  | 'GEOMETRYZM'
  | 'POINTZM'
  | 'LINESTRINGZM'
  | 'POLYGONZM'
  | 'MULTIPOINTZM'
  | 'MULTILINESTRINGZM'
  | 'MULTIPOLYGONZM'
  | 'GEOMETRYCOLLECTIONZM';

export type GeospatialPropertyType =
  | 'STRING'
  | 'LONG'
  | 'DOUBLE'
  | 'BOOLEAN'
  | 'TIMESTAMP'
  | GeometryPropertyType;

export type GeospatialFeatureTypeProperty =
  | {
      type: Exclude<GeospatialPropertyType, GeometryPropertyType | 'STRING'>;
      description?: string;
      optional?: boolean;
    }
  | {
      type: GeometryPropertyType;
      srid?: number;
      description?: string;
      optional?: boolean;
    }
  | {
      type: 'STRING';
      size: number;
      description?: string;
      optional?: boolean;
    };

type Properties = Record<string, GeospatialFeatureTypeProperty>;

export interface Geospatial {
  featureType: unknown;
  feature: unknown;
  crs: unknown;
}

export interface GeospatialFeature extends ExternalId {
  [property: string]: string | number | boolean | { wkt: string } | Geometry;
}

export interface GeospatialFeatureResponse extends GeospatialFeature {
  createdTime: number;
  lastUpdatedTime: number;
}

export interface FeatureType extends ExternalId, InternalId {
  properties: Properties & {
    createdTime: { type: 'LONG' };
    lastUpdatedTime: { type: 'LONG' };
    externalId: { type: 'STRING'; size: 32 };
  };
  searchSpec?: {
    [indexName: string]: GeospatialIndexSpec;
    createdTimeIdx: {
      properties: ['createdTime'];
    };
    lastUpdatedTimeIdx: {
      properties: ['lastUpdatedTime'];
    };
    externalIdIdx: {
      properties: ['externalId'];
    };
  };
}

type GeospatialIndexSpec = {
  properties: string[];
};

export interface GeospatialCreateFeatureType extends ExternalId {
  properties: Properties;
  searchSpec?: {
    [indexName: string]: GeospatialIndexSpec;
  };
}

export interface GeospatialUpdateFeatureType extends ExternalId {
  update:
    | {
        properties: { add: Properties } | { remove: string[] };
      }
    | {
        searchSpec:
          | {
              add: { [indexName: string]: { properties: string[] } };
            }
          | { remove: string[] };
      };
}

export interface GeospatialRecursiveDelete {
  recursive?: boolean;
}

export interface GeospatialOutput {
  output?:
    | {
        geometryFormat: GeometryType;
      }
    | {
        properties: Record<string, unknown>;
      };
}

interface GeospatialPropertyAndValue {
  property: string;
  value: unknown;
}

interface GeospatialPropertyAndPattern {
  property: string;
  pattern: string;
}

interface GeospatialAllowCrsTransformation {
  allowCrsTransformation?: boolean;
}

type GeospatialFeatureFilter =
  | {
      or: GeospatialFeatureFilter[];
    }
  | {
      and: GeospatialFeatureFilter[];
    }
  | {
      not: GeospatialFeatureFilter;
    }
  | {
      equals: GeospatialPropertyAndValue;
    }
  | {
      missing: { property: string };
    }
  | {
      like: GeospatialPropertyAndPattern;
    }
  | {
      regex: GeospatialPropertyAndPattern;
    }
  | {
      range:
        | { property: string; gt: string | number }
        | { property: string; lt: string | number }
        | { property: string; gte: string | number }
        | { property: string; lte: string | number };
    }
  | {
      stWithin: GeospatialPropertyAndValue;
    }
  | {
      stIntersects: GeospatialPropertyAndValue;
    }
  | {
      stContainsProperly: GeospatialPropertyAndValue;
    }
  | {
      stWithinDistance: GeospatialPropertyAndValue & { distance: number };
    };

export interface GeospatialFeatureSearchFilter
  extends GeospatialAllowCrsTransformation,
    Limit,
    GeospatialOutput {
  filter?: GeospatialFeatureFilter;
  sort?: string[];
}

export interface GeospatialFeatureSearchStreamFilter
  extends GeospatialAllowCrsTransformation,
    Limit {
  filter?: GeospatialFeatureFilter;
  output?:
    | GeospatialOutput['output']
    | {
        jsonStreamFormat:
          | 'LENGTH_PREFIXED'
          | 'NEW_LINE_DELIMITED'
          | 'CONCATENATED'
          | 'RECORD_SEPARATOR_DELIMITED';
      };
}

export interface FeatureAggregateParams {
  filter?: GeospatialFeatureFilter;
  aggregates: Aggregates[];
  property: string;
  outputSrid?: number;
  groupBy?: string[];
}

type Aggregates =
  | 'avg'
  | 'count'
  | 'max'
  | 'min'
  | 'stCentroid'
  | 'stCollect'
  | 'stConvexHull'
  | 'stIntersection'
  | 'stUnion'
  | 'sum'
  | 'variance';

export interface GeospatialCoordinateReferenceSystem {
  srid: number;
  wkt: string;
  projString: string;
}

export interface GeospatialCRSResponse
  extends GeospatialCoordinateReferenceSystem {
  createdTime: number;
  lastUpdatedTime: number;
}
