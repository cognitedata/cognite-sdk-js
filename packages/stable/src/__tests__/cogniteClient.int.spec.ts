// Copyright 2020 Cognite AS

import nock from 'nock';
import CogniteClient from '../cogniteClient';
import { Asset, ItemsWrapper } from '../types';
import {
  mockBaseUrl,
  setupLoggedInClient,
  setupMockableClient,
} from './testUtils';

describe('createClientWithApiKey - integration', () => {
  test('handle non-exisiting api-key', async () => {
    const client = new CogniteClient({
      appId: 'JS Integration test',
    });
    client.loginWithApiKey({
      project: 'cognitesdk-js',
      apiKey: 'non-exisiting-api-key',
    });
    await expect(
      client.assets.list({ limit: 1 }).autoPagingToArray({ limit: 1 })
    ).rejects.toThrowErrorMatchingInlineSnapshot(
      `"Request failed | status code: 401"`
    );
  });
});

describe('http methods - integration', () => {
  let client: CogniteClient;
  beforeAll(async () => {
    client = setupLoggedInClient();
  });
  test('post method', async () => {
    const assets = [{ name: 'First asset' }, { name: 'Second asset' }];
    const response = await client.post<ItemsWrapper<Asset[]>>(
      '/api/v1/projects/cognitesdk-js/assets',
      { data: { items: assets } }
    );
    expect(response.data.items).toHaveLength(2);
  });
  test('get method', async () => {
    const response = await client.get('/api/v1/projects/cognitesdk-js/assets');
    expect(response.data).toHaveProperty('items');
  });
});

describe('api endpoints smoke test', () => {
  let client: CogniteClient;

  beforeEach(() => {
    nock.cleanAll();
    client = setupMockableClient();
    const emptyResponse = { items: [] };
    nock(mockBaseUrl)
      .post(/.*/)
      .times(Infinity)
      .reply(200, emptyResponse);
    nock(mockBaseUrl)
      .get(/.*/)
      .times(Infinity)
      .reply(200, emptyResponse);
    nock(mockBaseUrl)
      .put(/.*/)
      .once()
      .reply(200, emptyResponse);
  });

  test('call (some) endpoints with a null context', async () => {
    async function callApi(api: any) {
      return Promise.all([
        callEndpoint(api.create, []),
        callEndpoint(api.list, {}),
        callEndpoint(api.retrieve, []),
        callEndpoint(api.update, []),
        callEndpoint(api.search, []),
        callEndpoint(api.delete, []),
        callEndpoint(api.insert, []),
      ]);
    }

    async function callEndpoint(endpoint: (a: any[]) => any, param: any) {
      if (endpoint) {
        const mockFn = jest.fn();
        await endpoint
          .bind(null)(param)
          .catch(mockFn);
        expect(mockFn).not.toBeCalled();
      }
    }

    await Promise.all(
      [
        client.assets,
        client.apiKeys,
        client.assets,
        client.datapoints,
        client.events,
        client.files,
        client.files3D,
        client.groups,
        client.login,
        client.logout,
        client.models3D,
        client.projects,
        client.raw,
        client.securityCategories,
        client.serviceAccounts,
        client.timeseries,
        client.viewer3D,
      ].map(callApi)
    );
  });
});
