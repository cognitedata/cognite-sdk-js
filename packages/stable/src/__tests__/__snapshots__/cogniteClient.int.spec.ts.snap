// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createClientWithOidc - integration handle login with existing OIDC credentials 1`] = `
Object {
  "capabilities": Array [
    Object {
      "groupsAcl": Object {
        "actions": Array [
          "CREATE",
          "DELETE",
          "UPDATE",
          "LIST",
          "READ",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
    },
    Object {
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
      "projectsAcl": Object {
        "actions": Array [
          "UPDATE",
          "LIST",
          "READ",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
    },
    Object {
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
      "threedAcl": Object {
        "actions": Array [
          "READ",
          "CREATE",
          "UPDATE",
          "DELETE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
    },
    Object {
      "assetsAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
    },
    Object {
      "eventsAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
    },
    Object {
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
      "securityCategoriesAcl": Object {
        "actions": Array [
          "MEMBEROF",
          "LIST",
          "CREATE",
          "UPDATE",
          "DELETE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
    },
    Object {
      "assetsAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
    },
    Object {
      "assetsAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
    },
    Object {
      "eventsAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
    },
    Object {
      "filesAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
    },
    Object {
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
      "rawAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
          "LIST",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
    },
    Object {
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
      "relationshipsAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
    },
    Object {
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
      "seismicAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
    },
    Object {
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
      "sequencesAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
    },
    Object {
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
      "timeSeriesAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
    },
    Object {
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
      "typesAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
    },
    Object {
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
      "templateGroupsAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
    },
    Object {
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
      "templateInstancesAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
    },
    Object {
      "datasetsAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
          "OWNER",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
    },
    Object {
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
      "securityCategoriesAcl": Object {
        "actions": Array [
          "MEMBEROF",
          "LIST",
          "CREATE",
          "UPDATE",
          "DELETE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
    },
    Object {
      "apikeysAcl": Object {
        "actions": Array [
          "LIST",
          "READ",
          "CREATE",
          "UPDATE",
          "DELETE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
    },
    Object {
      "groupsAcl": Object {
        "actions": Array [
          "LIST",
          "READ",
          "CREATE",
          "UPDATE",
          "DELETE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
    },
    Object {
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
      "usersAcl": Object {
        "actions": Array [
          "LIST",
          "READ",
          "CREATE",
          "UPDATE",
          "DELETE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
    },
    Object {
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
      "projectsAcl": Object {
        "actions": Array [
          "LIST",
          "READ",
          "UPDATE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
    },
    Object {
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
      "sessionsAcl": Object {
        "actions": Array [
          "LIST",
          "CREATE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
    },
    Object {
      "extractionPipelinesAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
    },
    Object {
      "extractionRunsAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
    },
    Object {
      "digitalTwinAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
    },
    Object {
      "modelHostingAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
    },
    Object {
      "genericsAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
    },
    Object {
      "functionsAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
    },
    Object {
      "labelsAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
    },
    Object {
      "geospatialAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
    },
    Object {
      "entitymatchingAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
    },
    Object {
      "documentPipelinesAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
    },
    Object {
      "filePipelinesAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
    },
    Object {
      "documentFeedbackAcl": Object {
        "actions": Array [
          "CREATE",
          "READ",
          "DELETE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
    },
    Object {
      "filePipelinesAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
    },
    Object {
      "annotationsAcl": Object {
        "actions": Array [
          "WRITE",
          "READ",
          "SUGGEST",
          "REVIEW",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
    },
    Object {
      "projectScope": Object {
        "projects": Array [
          "sdkcognite",
        ],
      },
      "wellsAcl": Object {
        "actions": Array [
          "READ",
          "WRITE",
        ],
        "scope": Object {
          "all": Object {},
        },
        "version": 1,
      },
    },
  ],
  "projects": Array [
    Object {
      "groups": Array [
        4688006025349041,
      ],
      "projectUrlName": "sdkcognite",
    },
  ],
  "subject": "6c5d3548-bf39-4ecc-b9ed-1c676a152780",
}
`;
